st2actions/st2actions/worker.py

```diff
@@ -134,7 +138,21 @@ def process(self, liveaction):
 
     def shutdown(self):
         super(ActionExecutionDispatcher, self).shutdown()
-        # Abandon running executions if incomplete
+
+        # Check if there are other runners in service registry
+        coordinator = coordination.get_coordinator()
+        member_ids = []
+        service = "actionrunner"
+
+        try:
```

Comment:
Is there a potential for race condition where, do we need locking?

Reply:
Moved refresh inside the loop